### PR TITLE
fix(sandpack-client): setup build with rollup

### PR DESCRIPTION
**Pull request by jeetiss**
I managed to repeat the build with rollup and it should fix the problem with code duplication

other improvements:

- `consoleHook` is smaller (terser with rollup `16393` vs esbuild `17253`)
- i migrate from the strategy "polyfill node globals" to direct polyfill usage

### CODE REVIEW

File: sandpack-client/src/clients/node/index.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 /* eslint-disable no-console,@typescript-eslint/no-explicit-any,prefer-rest-params,@typescript-eslint/explicit-module-boundary-types */
 
+import { Buffer } from "buffer";
```
Comment (jeetiss) [inline sandpack-client/src/clients/node/index.ts:3]: @danilowoz i'm not super familiar with the code base, could you approve that `buffer` is only one dependency that was polyfill with `@esbuild-plugins/node-globals-polyfill`? i do not see other things

File: sandpack-client/package.json [hunk 2]
```diff
@@ -31,7 +35,7 @@
     "prebuild": "yarn run clean",
     "test": "jest .",
     "lint": "eslint '**/*.ts?(x)' --fix",
-    "build": "node build.js && yarn run build:types",
+    "build": "rollup -c --bundleConfigAsCjs && yarn run build:types",
```
Comment (danilowoz) [inline sandpack-client/package.json:38]: Maybe we can let rollup generate the types and remove the `build:types`?

File: sandpack-react/rollup.config.js [hunk 3]
```diff
@@ -25,11 +25,14 @@ const configBase = {
 
   plugins: [
     typescript({ tsconfig: "./tsconfig.json" }),
-    replace({ "process.env.TEST_ENV": "false" }),
+    replace({
+      preventAssignment: true,
```
Comment (danilowoz) [inline sandpack-react/rollup.config.js:29]: Why this change is relevant for this PR?

File: sandpack-client/src/clients/node/index.ts (hunk 1 above)
Comment (danilowoz) [reply sandpack-client/src/clients/node/index.ts:3]: I believe so

**Review by danilowoz — COMMENTED** [review]
Could you please update your branch with `main` branch? I introduced important changes there.

File: sandpack-client/package.json (hunk 2 above)
Comment (jeetiss) [reply sandpack-client/package.json]: Yes, it should work

File: sandpack-react/rollup.config.js (hunk 3 above)
Comment (jeetiss) [reply sandpack-react/rollup.config.js:29]: i migrated `rollup-plugin-replace` to `@rollup/plugin-replace` in monorepo

`preventAssignment` flag is new in this plugin, it doesn't replace left values in the assignment to prevent bugs.

**Review by danilowoz — APPROVED** [review]
Let's get this merged! Thank you
